core/src/services/s3/core.rs

```diff
@@ -343,6 +344,31 @@ impl S3Core {
                 req = req.header(format!("{X_AMZ_META_PREFIX}{key}"), value)
             }
         }
+
+        // Set object tagging headers if tagging is enabled and tags are provided.
+        if !self.disable_tagging {
+            if let Some(tags) = args.tags() {
+                if !tags.is_empty() {
+                    let tags_string = tags
+                        .iter()
+                        .map(|(k, v)| {
+                            format!(
+                                "{}={}",
+                                k.replace('&', "%26")
+                                    .replace('=', "%3D")
+                                    .replace('+', "%2B"),
+                                v.replace('&', "%26")
+                                    .replace('=', "%3D")
+                                    .replace('+', "%2B")
```

Comment:
Could you provide the source of this logic? I suppose that S3's docs or other materials define it.